src/tasks/WasmAppBuilder/EmccExec.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Tasks;
+
+/// <summary>
+/// Wrapper around 'emcc' exec command that enrich ouput with msbuild configuration for produced errors.
+/// </summary>
+public class EmccExec : Exec
+{
+    public EmccExec()
+    {
+        ConsoleToMSBuild = true;
+    }
+
+    public override bool Execute()
+    {
+        bool result = base.Execute();
+
+        if (ExitCode != 0 && ConsoleOutput.Any(m => m.ItemSpec.Contains("wasm-ld: error:")) && ConsoleOutput.Any(m => m.ItemSpec.Contains("undefined symbol")))
+        {
+            Log.LogWarning("Use '-p:WasmAllowUndefinedSymbols=true' to allow undefined symbols");
```

Comment:
I'm logging it as warning, as it pops out visually and also I don't think it's actually another error.

Reply:
You can instead override `LogEventsFromTextOutput`.

Reply:
*If* we use a warning here, then it should have a warning code so users can ignore if it needed. Especially because it can become an error with warnAsError.

Reply:
Our warning is just after emscripten error. Turning of the warning doesn't make sense, since the original error will still be there.

Reply:
Then maybe we should be emitting an error too?

Reply:
I don't have strong opinion. I choosed warning mostly because it was better visible beside the error. I can change it to error if you prefer that.

Reply:
Changed to error.